control-api/run.go

```diff
@@ -111,6 +115,14 @@ func (request *DeployRequest) Validate() (*jwt.GenericClaims, error) {
 		return nil, fmt.Errorf("workload name claim ('%s') does not match requirements (%s)", claims.Subject, workloadRegex)
 	}
 
+	if request.Essential != nil && *request.Essential && request.WorkloadType != NexWorkloadNative {
+		return nil, errors.New("essential workloads must be native")
+	}
+
+	if !strings.EqualFold(request.Location.Scheme, WorkloadLocationSchemeFile) && !strings.EqualFold(request.Location.Scheme, WorkloadLocationSchemeNATS) {
```

Comment:
I can see we're validating that the scheme is correct.. but where do we validate that a `file://` workload is only supported by firecracker sandboxes?

Reply:
https://github.com/synadia-io/nex/pull/305/files#diff-019cc1c548465bdfa17427d6a8f9a2039c705a13c83a04712330d22658e0ad1aR161-R163

👍🏻

Reply:
In addition to the workload manager checking to ensure we are in a sandbox, the agent also prevents execution: https://github.com/synadia-io/nex/pull/305/files#diff-62b6ad581fe3a3059ae8c85ef0f31dde4092bfdecfa7d6857c470bcacaa8cc8bR185-R187